solc/CommandLineParser.cpp

```diff
@@ -1460,6 +1489,61 @@ void CommandLineParser::processArgs()
 		m_options.input.mode == InputMode::CompilerWithASTImport ||
 		m_options.input.mode == InputMode::EVMAssemblerJSON
 	);
+
+	std::string ethdebugOutputSelection =
+		"--" + CompilerOutputs::componentName(&CompilerOutputs::ethdebug) + " / --" + CompilerOutputs::componentName(&CompilerOutputs::ethdebugRuntime);
+
+	bool incompatibleEthdebugOutputs =
+		m_options.compiler.outputs.asmJson || m_options.compiler.outputs.irAstJson || m_options.compiler.outputs.irOptimizedAstJson;
+
+	bool incompatibleEthdebugInputs = m_options.input.mode != InputMode::Compiler;
+
+	if (m_options.compiler.outputs.ethdebug || m_options.compiler.outputs.ethdebugRuntime)
+	{
+		if (!m_options.output.viaIR)
+			solThrow(
+				CommandLineValidationError,
+				"--" + CompilerOutputs::componentName(&CompilerOutputs::ethdebug) + " / --" + CompilerOutputs::componentName(&CompilerOutputs::ethdebugRuntime) + " output can only be selected, if --via-ir was specified."
+			);
+
+		if (incompatibleEthdebugOutputs)
+			solThrow(
+				CommandLineValidationError,
+				ethdebugOutputSelection + " output can only be used with --" + CompilerOutputs::componentName(&CompilerOutputs::ir) +
```

Comment:
re-use `ethdebugOutputSelection`

Reply:
its already reused here, but I guess you thought that `CompilerOutputs::componentName(&CompilerOutputs::ir)...` is related `ethdebugOutputSelection` - however, it make sense to also create a specific message for that ir stuff